sphinx/config.py

```diff
@@ -161,11 +164,37 @@ def __init__(self, config: Dict[str, Any] = {}, overrides: Dict[str, Any] = {})
                 config['extensions'] = self.overrides.pop('extensions')
         self.extensions: List[str] = config.get('extensions', [])
 
+    @classmethod
+    def get_config_file(cls, confdir: str) -> str:
```

Comment:
What purpose is the return value of this function? I think the filename (and filetype) of config should be encapsulated to this class. So it would be better to rename this as `Config.exists?(confdir)` instead.

Or it's enough to raise an error on reading instead of checking existence.

Reply:
see above 👆